src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -115,6 +119,58 @@ public ProgrammingExerciseGradingService(ProgrammingExerciseTestCaseService test
                 triggerTemplateBuildIfTestCasesChanged(programmingExercise.getId(), result.getId());
             }
+
+            if (!isSolutionParticipation && !isTemplateParticipation) {
+
+                // check if a manual result exists, if so we want to clone and update this with the new updated test-case and sca feedback
+                Optional<ProgrammingExerciseStudentParticipation> studentParticipation = programmingExerciseParticipationService
+                        .findStudentParticipationWithLatestManualResultsAndFeedbacksAndRelatedSubmissionsAndAssessor(participation.getId());
+
+                if (studentParticipation.isPresent() && studentParticipation.get().getResults().stream().findFirst().isPresent()) {
+
+                    Result manualResult = studentParticipation.get().getResults().stream().findFirst().get();
+
+                    // create a new manual result for the same submission as the automatic result
+                    Result newResult = programmingSubmissionService.saveNewEmptyResult(result.getSubmission());
+
+                    newResult.setAssessor(manualResult.getAssessor());
+                    newResult.setAssessmentType(AssessmentType.SEMI_AUTOMATIC);
+                    // this makes it the most recent result, but optionally keeps the draft state of an unfinished manual result
+                    newResult.setCompletionDate(manualResult.getCompletionDate() != null ? result.getCompletionDate().plusSeconds(1) : null);
+                    newResult.setHasFeedback(manualResult.getHasFeedback());
+                    newResult.setRated(manualResult.isRated());
+
+                    // copy all feedback from the automatic result
+                    for (Feedback feedback : result.getFeedbacks()) {
+                        Feedback newFeedback = feedback.copyFeedback();
+                        newResult.addFeedback(newFeedback);
+                    }
+
+                    // copy only the non-automatic feedback from the manual result
+                    for (Feedback feedback : manualResult.getFeedbacks()) {
+                        if (feedback != null && feedback.getType() != FeedbackType.AUTOMATIC) {
+                            Feedback newFeedback = feedback.copyFeedback();
+                            newResult.addFeedback(newFeedback);
+                        }
+                    }
+
+                    // recalculate the score and set the result string
+                    double maxScore = getMaxScoreRespectingZeroPointExercises(programmingExercise);
+                    double points = programmingAssessmentService.calculateTotalScore(newResult);
+                    newResult.setScore(points, maxScore);
+
+                    String resultString = result.getResultString() + ", " + newResult.createResultString(points, maxScore);
+                    newResult.setResultString(resultString);
```

Comment:
We have similar statements in updateResultString(). We could factor this out into a new method to avoid code duplication.

Reply:
Done